resources/sdk/purecloudjavascript/templates/configuration.mustache

```diff
@@ -0,0 +1,128 @@
+const ConfigParser = require('configparser');
+const os = require('os');
+const path = require('path');
+const fs = require('fs');
+
+import Logger from './logger.js';
+
+class Configuration {
+	/**
+	 * Singleton getter
+	 */
+	get instance() {
+		return Configuration.instance;
+	}
+
+	/**
+	 * Singleton setter
+	 */
+	set instance(value) {
+		Configuration.instance = value;
+	}
+
+	constructor() {
+		if (!Configuration.instance) {
+			Configuration.instance = this;
+		}
+
+		this.configPath = typeof window !== 'undefined' ? '' : path.join(os.homedir(), '.genesyscloudjavascript', 'config');
+		this.refresh_access_token = true;
+		this.refresh_token_wait_max = 10;
+		this.live_reload_config = true;
+		this.host;
+		this.environment;
+		this.basePath;
+		this.authUrl;
+		this.config;
+		this.logger = new Logger();
+		this.setEnvironment();
+		this.liveLoadConfig();
+	}
+
+	liveLoadConfig() {
+		// If in browser, don't read config file, use default values
+		if (typeof window !== 'undefined') {
+			this.configPath = '';
+			return;
+		}
+
+		if (!fs.existsSync(this.configPath)) {
+			console.error('No config file exists in the specified path.');
+			return;
+		}
+
+		this.updateConfigFromFile();
+
+		if (this.live_reload_config && this.live_reload_config === true) {
+			const watcher = fs.watch(this.configPath, (eventType, filename) => {
+				this.updateConfigFromFile();
+				if (!this.live_reload_config) {
+					watcher.close();
+				}
+			});
+		}
+	}
+
+	setConfigPath(path) {
+		if (path) {
+			this.configPath = path;
+		}
+	}
+
+	updateConfigFromFile() {
+		var configparser = new ConfigParser();
+
+		try {
+			configparser.read(this.configPath); // If no error catched, indicates it's INI format
+			this.config = configparser;
+		} catch (error) {
+			if (error.name && error.name === 'MissingSectionHeaderError') {
+				// Not INI format, see if it's JSON format
+				var configData = fs.readFileSync(this.configPath, 'utf8');
+				this.config = {
+					_sections: JSON.parse(configData), // To match INI data format
+				};
+			} else {
+				console.error(error);
+				return;
+			}
+		}
+
+		if (this.config) this.updateConfigValues();
+	}
+
+	updateConfigValues() {
```

Comment:
You're going to have to follow the same pattern I applied for reading config in the other SDKs.
The config file should be able to have all possible values, or only one if the user wishes.
I tried the following file and it errors out
{
    "logging": {
        "log_level": "trace"
    }
}